Core/Core/SwiftUIViews/PrimaryButton.swift

```diff
@@ -0,0 +1,53 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public struct PrimaryButton<Label>: View where Label: View {
+
+    let action: () -> Void
+    let label: Label
+    @State private var shouldShowAlert = false
+    @ObservedObject var offlineServiceModel = OfflineServiceModel(offlineService: OfflineServiceLive.shared)
```

Comment:
This is a dependency owned by the view so it should be `@StateObject`. Also it can be `private`.
```suggestion
    @StateObject private var offlineServiceModel = OfflineServiceModel(offlineService: OfflineServiceLive.shared)
```

Reply:
done.